cache/refs.go

```diff
@@ -201,6 +212,15 @@ func (cr *cacheRecord) parentRef(hidden bool) *immutableRef {
 }
 
 func (cr *cacheRecord) Mount(ctx context.Context, readonly bool) (snapshot.Mountable, error) {
+	if getBlobOnly(cr.md) {
```

Comment:
Yes, will do. It's just because we have to create an immutable ref in order to be able to call the extract method here, but since we don't return that immutable ref we have to release it or else it'll leak.

Now that I type that out, I wonder if it's possible to just define the `extract` method on `cacheRecord` instead of `immutableRef`, which could avoid this issue entirely. I'll give that a shot and if it doesn't work out update with the comment.

Reply:
Tried this, while it's possible to move around the methods to avoid getting a temporary ref, it causes you to migrate a ton of code from  `*immutableRef` to `*cacheRecord` that really should only be defined for `*immutableRef`, all only to support this one line. The overhead of getting a temporary ref seems to be quite small, so I am going with that for now and added a comment like you suggested.